src/pages/settings/task-statuses/components/CreateTaskStatusModal.tsx

```diff
@@ -0,0 +1,124 @@
+ * @link https://github.com/invoiceninja/invoiceninja source repository
+ *
+ * @copyright Copyright (c) 2022. Invoice Ninja LLC (https://invoiceninja.com)
+ *
+ * @license https://www.elastic.co/licensing/elastic-license
+ */
+
+import { Modal } from 'components/Modal';
+import { Dispatch, SetStateAction } from 'react';
+import { Button, InputField, InputLabel } from '@invoiceninja/forms';
+import { ColorPicker } from 'components/forms/ColorPicker';
+import { AxiosError } from 'axios';
+import { endpoint } from 'common/helpers';
+import { request } from 'common/helpers/request';
+import { toast } from 'common/helpers/toast/toast';
+import { useAccentColor } from 'common/hooks/useAccentColor';
+import { useTitle } from 'common/hooks/useTitle';
+import { TaskStatus } from 'common/interfaces/task-status';
+import { ValidationBag } from 'common/interfaces/validation-bag';
+import { FormEvent, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from 'react-query';
+import { useHandleChange } from '../common/hooks';
+import { useBlankTaskStatusQuery } from 'common/queries/task-statuses';
+
+interface Props {
+  visible: boolean;
+  setVisible: Dispatch<SetStateAction<boolean>>;
+}
+
+export function CreateTaskStatusModal(props: Props) {
+  useTitle('kanban');
+
+  const [t] = useTranslation();
+
+  const accentColor = useAccentColor();
+
+  const queryClient = useQueryClient();
+
+  const { data: blankTaskStatus } = useBlankTaskStatusQuery();
+
+  const [errors, setErrors] = useState<ValidationBag>();
+
+  const [isFormBusy, setIsFormBusy] = useState<boolean>(false);
+
+  const [taskStatus, setTaskStatus] = useState<TaskStatus>();
+
+  const handleChange = useHandleChange({
+    setErrors,
+    setTaskStatus,
+  });
+
+  const handleSave = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!isFormBusy) {
+      toast.processing();
+
+      setErrors(undefined);
+
+      setIsFormBusy(true);
+
+      request('POST', endpoint('/api/v1/task_statuses'), taskStatus)
+        .then(() => {
+          toast.success('created_task_status');
+
+          queryClient.invalidateQueries('/api/v1/task_statuses');
+
+          setTaskStatus(blankTaskStatus);
+
+          props.setVisible(false);
+        })
+        .catch((error: AxiosError<ValidationBag>) => {
+          toast.dismiss();
+          console.error(error);
+
+          error.response?.status === 422
+            ? setErrors(error.response.data)
+            : toast.error();
+        })
+        .finally(() => setIsFormBusy(false));
+    }
+  };
+
+  useEffect(() => {
+    if (blankTaskStatus) {
+      setTaskStatus(blankTaskStatus);
+    }
+  }, [blankTaskStatus]);
+
+  return (
+    <Modal
+      title={t('new_task_status')}
+      visible={props.visible}
+      onClose={() => {
+        props.setVisible(false);
+        setTaskStatus(blankTaskStatus);
+      }}
+    >
+      <div className="space-y-6 px-4">
```

Comment:
There's no need for this spacing. Please keep the one that's default to all modals.

Reply:
Sure, removed!